jruby/neo4j/driver/ext/internal/summary/internal_result_summary.rb

```diff
@@ -9,6 +9,21 @@ module InternalResultSummary
             %i[result_available_after result_consumed_after].each do |method|
               define_method(method) { super(Java::JavaUtilConcurrent::TimeUnit::MILLISECONDS) }
             end
+
+            def self.query_type(type)
+              case type
+              when "READ_ONLY"
+                'r'
+              when "READ_WRITE"
+                'rw'
+              when "WRITE_ONLY"
+                'w'
+              when "SCHEMA_WRITE"
+                's'
+              else
+                raise Neo4j::Driver::Exceptions::ClientException, "Unknown query type: #{type}"
```

Comment:
I would skip the `else`. The way QueryType is constructed in java driver, an exception would have already been raised if nothing matched. So this line is unreachable anyway and only will show up as missing test coverage.